pkg/cmd/factory/remote_resolver.go

```diff
@@ -59,21 +59,36 @@ func (rr *remoteResolver) Resolver() func() (context.Remotes, error) {
 		var hostname string
 		cachedRemotes = context.Remotes{}
 		sort.Sort(resolvedRemotes)
-		for _, r := range resolvedRemotes {
-			if hostname == "" {
-				if !knownHosts[r.RepoHost()] {
+
+		if ghinstance.Default() != ghinstance.OverridableDefault() {
```

Comment:
Instead of coupling this logic to `ghinstance`, could the `Resolver()` method accept a `hostname` argument that, if non-blank, explicitly chooses a hostname? Then, `ghinstance` could be consulted at the call site of `Resolver()`.

That way, Resolver is more easily testable, and its caller is responsible for wiring up the relevant parts of the user's environment.